src/components/filter/FilterPanel.tsx

```diff
@@ -72,7 +72,8 @@ const FilterPanel = (props: Props) => {
 	}, [showFilter]);
 
 	useOutsideClickListener(filterBaseRef, (e: MouseEvent) => {
-		if (!filterButtonRef.current?.contains(e.target as Element)) {
+		const fromFiltersHistory = Boolean((e.target as HTMLElement).closest('.filters-history'));
```

Comment:
wrap renderFooter with div, give it a ref, check if this footer ref include click target.